### PR TITLE
Remove REPL dependency

### PR DESCRIPTION
**Pull request by topolarity**
For non-interactive (e.g. script or executable-like) workflows, REPL is a pretty heavy dependency that touches a lot of global state.

Avoid the dependency by copying the `levenshtein` implementation instead.

### CODE REVIEW

File: src/other/index.jl [hunk 1]
```diff
@@ -294,6 +294,25 @@ end
 @inline Base.getindex(x::AbstractIndex, rx::Regex) =
     getindex(x, filter(name -> occursin(rx, String(name)), _names(x)))
 
+# Levenshtein Distance
```
Comment (bkamins) [inline src/other/index.jl:297]: please add a note about the source

File: src/other/index.jl [hunk 2]
```diff
@@ -294,6 +294,25 @@ end
...
+function levenshtein(s1, s2)
```
Comment (bkamins) [inline src/other/index.jl:298]:
```suggestion
function _levenshtein(s1, s2)
```

File: src/other/index.jl [hunk 3]
```diff
@@ -302,7 +321,7 @@ end
 # Returns candidates ordered by (distance, name) pair
 function fuzzymatch(l::Dict{Symbol, Int}, idx::Symbol)
         idxs = uppercase(string(idx))
-        dist = [(REPL.levenshtein(uppercase(string(x)), idxs), x) for x in keys(l)]
+        dist = [(levenshtein(uppercase(string(x)), idxs), x) for x in keys(l)]
```
Comment (bkamins) [inline src/other/index.jl:324]:
```suggestion
        dist = [(_levenshtein(uppercase(string(x)), idxs), x) for x in keys(l)]
```

**Review by bkamins — COMMENTED** [review]
Looks good. I left some minor comments

File: src/other/index.jl (hunk 1 above)
Comment (topolarity) [reply src/other/index.jl:297]: You can find pretty identical pseudocode all over the internet (e.g. https://en.wikipedia.org/wiki/Levenshtein_distance#Iterative_with_full_matrix) but will do

Comment (bkamins) [reply src/other/index.jl:297]: Thank you. This is relevant for making sure that the whole code of DataFrames.jl can be distributed under MIT license.

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**